src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -63,39 +64,72 @@ import {
   fetchExportObjects,
   fetchExportByTypeAndSearch,
   findObjects,
+  extractExportDetails,
+  SavedObjectsExportResultDetails,
 } from '../../lib';
-import { extractExportDetails } from '../../lib/extract_export_details';
-
-export const POSSIBLE_TYPES = chrome.getInjected('importAndExportableTypes');
-
-export class ObjectsTable extends Component {
-  static propTypes = {
-    savedObjectsClient: PropTypes.object.isRequired,
-    indexPatterns: PropTypes.object.isRequired,
-    $http: PropTypes.func.isRequired,
-    basePath: PropTypes.string.isRequired,
-    perPageConfig: PropTypes.number,
-    newIndexPatternUrl: PropTypes.string.isRequired,
-    confirmModalPromise: PropTypes.func.isRequired,
-    services: PropTypes.array.isRequired,
-    uiCapabilities: PropTypes.object.isRequired,
-    goInspectObject: PropTypes.func.isRequired,
-    canGoInApp: PropTypes.func.isRequired,
-  };
+import { SavedObjectWithMetadata } from '../../types';
+import {
+  ISavedObjectsManagementServiceRegistry,
+  SavedObjectsManagementActionServiceStart,
+} from '../../services';
+import { Header, Table, Flyout, Relationships } from './components';
+
+interface ExportAllOption {
+  id: string;
+  label: string;
+}
 
-  constructor(props) {
+export interface SavedObjectsTableProps {
+  allowedTypes: string[];
+  serviceRegistry: ISavedObjectsManagementServiceRegistry;
+  actionRegistry: SavedObjectsManagementActionServiceStart;
+  savedObjectsClient: SavedObjectsClientContract;
+  indexPatterns: IndexPatternsContract;
+  http: HttpStart;
+  overlays: OverlayStart;
+  notifications: NotificationsStart;
+  applications: ApplicationStart;
+  perPageConfig: number;
+  goInspectObject: (obj: SavedObjectWithMetadata) => void;
+  canGoInApp: (obj: SavedObjectWithMetadata) => boolean;
+}
+
+export interface SavedObjectsTableState {
+  totalCount: number;
+  page: number;
+  perPage: number;
+  savedObjects: SavedObjectWithMetadata[];
+  savedObjectCounts: Record<string, number>;
+  activeQuery: Query;
+  selectedSavedObjects: SavedObjectWithMetadata[];
+  isShowingImportFlyout: boolean;
+  isSearching: boolean;
+  filteredItemCount: number;
+  isShowingRelationships: boolean;
+  relationshipObject?: SavedObjectWithMetadata;
+  isShowingDeleteConfirmModal: boolean;
+  isShowingExportAllOptionsModal: boolean;
+  isDeleting: boolean;
+  exportAllOptions: ExportAllOption[];
+  exportAllSelectedOptions: Record<string, boolean>;
+  isIncludeReferencesDeepChecked: boolean;
+}
+
+export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedObjectsTableState> {
+  private _isMounted = false;
```

Comment:
🙈 on that one. This is there because `debouncedFetch` is performing debounced http calls followed by `setState` calls (which will raise an error when called after unmount). Removing that would require to be able to interrupt the http requests on unmount, which is indeed better and technically possible with NP apis but way more complicated than this ugly manual mount stats property (which is indeed an anti-pattern)...